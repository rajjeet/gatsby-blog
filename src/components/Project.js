import React from "react";
import styled from "styled-components";
import {navigate} from "gatsby";

function Project({header, description, link}) {
    return <a onClick={() => navigate(link)}>
        <h3>{header}</h3>
        <p>{description}</p>
    </a>;
}

const StyledProject = styled(Project)`
  display: inline-block;
        border: 1px solid gray;
        padding: 1em;
        margin-right: 1em;
        cursor: pointer;
        width: 40%;
        :hover {
            background-color: #eee;
            transform: translateY(-5px);
        }
        transition: transform .5s ease;
        h3 {
          margin-bottom: .2em;
        }
`;

export default StyledProject;