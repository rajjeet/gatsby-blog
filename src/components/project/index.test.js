import React from 'react';
import renderer from 'react-test-renderer';
import Project from './index';

const makeProps = () => ({
  heading: 'Sample Heading', link: '', description: '', className: {}, thumbnail: null,
});

describe('<Project />', () => {
  it.skip('should render', () => {
    const tree = renderer.create(<Project {...makeProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
