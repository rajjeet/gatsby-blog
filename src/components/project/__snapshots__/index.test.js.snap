// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should render 1`] = `
<DocumentFragment>
  <div
    class="[object Object] sc-htpNat cDaIEp"
  >
    <a
      aria-label="See Sample Project"
      class="sc-bwzfXH kTyWdp"
      href=""
    >
      <h3>
        Sample Project
      </h3>
      <div
        class="sc-bdVaJa cBDXGb gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%; padding-bottom: 100%;"
        />
        <picture>
          <source
            sizes=""
            srcset=""
          />
          <img
            alt="Sample Project"
            src=""
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;img src="" alt="Sample Project" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <p>
        This is a sample project description
      </p>
    </a>
  </div>
</DocumentFragment>
`;
